apps/signing/serializers.py

```diff
@@ -0,0 +1,81 @@
+import json
+
+from django.core.urlresolvers import reverse
+
+from rest_framework import serializers
+
+import amo
+from amo.helpers import absolutify
+from files.models import FileUpload
+
+
+class FileUploadSerializer(serializers.ModelSerializer):
+    active = serializers.SerializerMethodField('get_active')
+    api_url = serializers.SerializerMethodField('get_api_url')
+    files = serializers.SerializerMethodField('get_files')
+    passed_review = serializers.SerializerMethodField('get_passed_review')
+    processed = serializers.CharField(source='processed')
+    reviewed = serializers.SerializerMethodField('get_reviewed')
+    validation_results = serializers.SerializerMethodField(
+        'get_validation_results')
+    validation_url = serializers.SerializerMethodField('get_validation_url')
+
+    class Meta:
+        model = FileUpload
+        fields = [
+            'active',
+            'api_url',
+            'files',
+            'passed_review',
+            'processed',
+            'reviewed',
+            'valid',
+            'validation_results',
+            'validation_url',
+            'version',
+        ]
+
+    def __init__(self, *args, **kwargs):
+        self.version = kwargs.pop('version', None)
+        super(FileUploadSerializer, self).__init__(*args, **kwargs)
+
+    def get_api_url(self, instance):
+        return absolutify(reverse('signing.version', args=[instance.addon.guid,
+                                                           instance.version]))
+
+    def get_validation_url(self, instance):
+        return absolutify(reverse('devhub.upload_detail',
+                                  args=[instance.uuid]))
+
+    def get_files(self, instance):
+        if self.version is not None:
+            return [{'download_url': f.get_url_path('api')}
+                    for f in self.version.files.all()]
+        else:
+            return []
+
+    def get_validation_results(self, instance):
+        if instance.validation:
+            return json.loads(instance.validation)
+        else:
+            return None
+
+    def get_reviewed(self, instance):
+        if self.version is not None:
+            return all(f.reviewed for f in self.version.files.all())
+        else:
+            return False
+
+    def get_active(self, instance):
+        if self.version is not None:
+            return all(status[1] in [amo.STATUS_PUBLIC, amo.STATUS_LITE,
+                                     amo.STATUS_LITE_AND_NOMINATED]
+                       for status in self.version.statuses)
+        else:
+            return False
+
+    def get_passed_review(self, instance):
```

Comment:
could be a one liner: `return self.get_reviewed(instance) and self.get_active(instance)`